tests/Unit/Rule/ContextAspectValidationRule/Fixture/UseContextApiWithDefinedAspect.php

```diff
@@ -0,0 +1,35 @@
+<?php declare(strict_types = 1);
+
+namespace SaschaEgerer\PhpstanTypo3\Tests\Unit\Rule\ContextAspectValidationRule\Fixture;
+
+use TYPO3\CMS\Core\Context\Context;
+use TYPO3\CMS\Core\Context\DateTimeAspect;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+final class UseContextApiWithDefinedAspect
+{
+
+	public function someMethod(): void
+	{
+		$dateAspect = GeneralUtility::makeInstance(Context::class)->getAspect('date');
+		$dateAspect->get('bar');
+
+		$class = new class {
+
+			public function getAspect(string $name): \stdClass
+			{
+				return new \stdClass();
+			}
+
+		};
+
+		$myCustomAspect = $class->getAspect('foo');
+		$myCustomAspect->something = 'FooBarBaz';
+
+		$aspectWithoutDefiningName = GeneralUtility::makeInstance(Context::class)->getAspect();
```

Comment:
How is this possible? Should be an PHP type error?

Reply:
I just wanted to test that our rule does not care about it because no argument is given. This is an error which PHPStan should complaint about it. As we are statically analysing the code no Type Error is thrown. For the sake of our test, it should be fine.